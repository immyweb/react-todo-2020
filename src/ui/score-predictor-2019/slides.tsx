import * as React from 'react';

import Slide from './slide';

import { SlidesContainer, SlidesInner, Cta, SubmitCtaHolder } from './styles';
import { colours } from '../../styles/colours';

import { IFixture, ITally, ITeam, IIntro } from './types';
import { ITheme } from '../../types';

interface IProps {
  theme: ITheme;
  fixturesGrouped: IFixture[][];
  teams: ITeam[];
  currentPos: number;
  totalWidth: number;
  slideWidth: number;
  isLastSlide: boolean;
  intro: IIntro;
  onPredict(tally: ITally[]): void;
  onConfirm(): void;
}

const Slides: React.FC<IProps> = ({
  theme,
  fixturesGrouped,
  teams,
  currentPos,
  totalWidth,
  slideWidth,
  isLastSlide,
  intro,
  onPredict,
  onConfirm,
}) => {
  const slides = fixturesGrouped.map((group: IFixture[], i: number) => {
    return (
      <Slide
        key={i}
        group={group}
        teams={teams}
        theme={theme}
        slideWidth={slideWidth}
        onPredict={onPredict}
      />
    );
  });
  return (
    <SlidesContainer>
      <SlidesInner width={totalWidth} currentPos={currentPos}>
        {slides}
      </SlidesInner>
      {isLastSlide && (
        <SubmitCtaHolder>
          <Cta
            colour={{ colour: colours.$white, background: theme.primary }}
            hover={{ colour: colours.$white, background: theme.secondary }}
            onClick={onConfirm}
          >
            {intro.ctaResults}
          </Cta>
        </SubmitCtaHolder>
      )}
    </SlidesContainer>
  );
};

export default Slides;
