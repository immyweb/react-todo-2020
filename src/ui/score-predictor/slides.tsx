import * as React from 'react';
import glamorous from 'glamorous';

import { Slide } from './slide';
import { SlidesContainer } from './styles';

import { IFixture, ITeam, ITally } from './types';

interface IGProps {
  width?: number;
  currentPos?: number;
}

interface IProps {
  tables: IFixture[][];
  teams: ITeam[];
  currentSlide: number;
  currentPos: number;
  totalWidth: number;
  slideWidth: number;
  onPredict(tally: ITally[]): void;
  onPlayed(fixtureId: number): void;
  onToggleNext(): void;
}

export const renderSlides = (
  tables: IFixture[][],
  teams: ITeam[],
  onPredict: (tally: ITally[]) => void,
  slideWidth: number,
  onPlayed: (fixtureId: number) => void,
  onToggleNext: () => void,
) => {
  return tables.map((group: IFixture[], i: number) => {
    return (
      <Slide
        key={i}
        group={group}
        teams={teams}
        slideWidth={slideWidth}
        onPredict={onPredict}
        onPlayed={onPlayed}
        onToggleNext={onToggleNext}
      />
    );
  });
};

export const Slides: React.SFC<IProps> = ({
  tables,
  teams,
  currentSlide,
  currentPos,
  totalWidth,
  slideWidth,
  onPredict,
  onPlayed,
  onToggleNext,
}) => {
  const slides = renderSlides(tables, teams, onPredict, slideWidth, onPlayed, onToggleNext);

  return (
    <SlidesContainer>
      <SlidesInner width={totalWidth} currentPos={currentPos}>
        {slides}
      </SlidesInner>
    </SlidesContainer>
  );
};

const SlidesInner = glamorous.div<IGProps>(
  {
    label: 'slides-inner',
    transitionDuration: '.5s',
    transitionTimingFunction: 'ease',
    transitionProperty: 'all',
  },
  ({ width, currentPos }) => ({
    width: width ? width : '100%',
    transform: `translate3d(${currentPos ? -currentPos : 0}px, 0, 0)`,
  }),
);
