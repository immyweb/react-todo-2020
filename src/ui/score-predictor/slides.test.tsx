import * as React from 'react';
import { shallow } from 'enzyme';

import { Slides, renderSlides } from './slides';

const teams = [
  {
    team: 'Liverpool',
    badge: 'https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67',
    played: 31,
    win: 18,
    draw: 9,
    lose: 4,
    points: 63,
  },
  {
    team: 'C Palace',
    badge:
      'https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72',
    played: 31,
    win: 7,
    draw: 9,
    lose: 15,
    points: 30,
  },
  {
    team: 'Brighton',
    badge:
      'https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96',
    played: 30,
    win: 8,
    draw: 10,
    lose: 12,
    points: 34,
  },
  {
    team: 'Leicester',
    badge:
      'https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90',
    played: 30,
    win: 10,
    draw: 10,
    lose: 10,
    points: 40,
  },
];

const tables = [
  [
    { id: 0, team1: 'C Palace', team2: 'Liverpool' },
    { id: 1, team1: 'Brighton', team2: 'Leicester' },
    { id: 2, team1: 'C Palace', team2: 'Liverpool' },
    { id: 3, team1: 'Brighton', team2: 'Leicester' },
    { id: 4, team1: 'C Palace', team2: 'Liverpool' },
    { id: 5, team1: 'Brighton', team2: 'Leicester' },
    { id: 6, team1: 'C Palace', team2: 'Liverpool' },
    { id: 7, team1: 'Brighton', team2: 'Leicester' },
  ],
  [
    { id: 8, team1: 'C Palace', team2: 'Liverpool' },
    { id: 9, team1: 'Brighton', team2: 'Leicester' },
  ],
];

function shallowSlides(props?: any) {
  return shallow(
    <Slides
      tables={tables}
      teams={teams}
      currentSlide={0}
      currentPos={0}
      totalWidth={1240}
      slideWidth={620}
      onPredict={() => {}}
      onPlayed={() => {}}
      onToggleNext={() => {}}
      {...props}
    />,
  );
}

describe('<Slides />', () => {
  it('renders correctly', () => {
    const wrapper = shallowSlides();
    expect(wrapper).toMatchSnapshot();
  });

  it('renderSlides returns correct number of <Slide /> components', () => {
    const wrapper = renderSlides(tables, teams, () => {}, 620, () => {}, () => {});
    expect(wrapper.length).toEqual(2);
  });
});
