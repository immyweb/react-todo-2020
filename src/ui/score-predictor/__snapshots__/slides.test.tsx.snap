// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slides /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    currentPos={0}
    currentSlide={0}
    onPlayed={[Function]}
    onPredict={[Function]}
    onToggleNext={[Function]}
    slideWidth={620}
    tables={
      Array [
        Array [
          Object {
            "id": 0,
            "team1": "C Palace",
            "team2": "Liverpool",
          },
          Object {
            "id": 1,
            "team1": "Brighton",
            "team2": "Leicester",
          },
          Object {
            "id": 2,
            "team1": "C Palace",
            "team2": "Liverpool",
          },
          Object {
            "id": 3,
            "team1": "Brighton",
            "team2": "Leicester",
          },
          Object {
            "id": 4,
            "team1": "C Palace",
            "team2": "Liverpool",
          },
          Object {
            "id": 5,
            "team1": "Brighton",
            "team2": "Leicester",
          },
          Object {
            "id": 6,
            "team1": "C Palace",
            "team2": "Liverpool",
          },
          Object {
            "id": 7,
            "team1": "Brighton",
            "team2": "Leicester",
          },
        ],
        Array [
          Object {
            "id": 8,
            "team1": "C Palace",
            "team2": "Liverpool",
          },
          Object {
            "id": 9,
            "team1": "Brighton",
            "team2": "Leicester",
          },
        ],
      ]
    }
    teams={
      Array [
        Object {
          "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
          "draw": 9,
          "lose": 4,
          "played": 31,
          "points": 63,
          "team": "Liverpool",
          "win": 18,
        },
        Object {
          "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
          "draw": 9,
          "lose": 15,
          "played": 31,
          "points": 30,
          "team": "C Palace",
          "win": 7,
        },
        Object {
          "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
          "draw": 10,
          "lose": 12,
          "played": 30,
          "points": 34,
          "team": "Brighton",
          "win": 8,
        },
        Object {
          "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
          "draw": 10,
          "lose": 10,
          "played": 30,
          "points": 40,
          "team": "Leicester",
          "win": 10,
        },
      ]
    }
    totalWidth={1240}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <glamorous(div)
        currentPos={0}
        width={1240}
      >
        <Slide
          group={
            Array [
              Object {
                "id": 0,
                "team1": "C Palace",
                "team2": "Liverpool",
              },
              Object {
                "id": 1,
                "team1": "Brighton",
                "team2": "Leicester",
              },
              Object {
                "id": 2,
                "team1": "C Palace",
                "team2": "Liverpool",
              },
              Object {
                "id": 3,
                "team1": "Brighton",
                "team2": "Leicester",
              },
              Object {
                "id": 4,
                "team1": "C Palace",
                "team2": "Liverpool",
              },
              Object {
                "id": 5,
                "team1": "Brighton",
                "team2": "Leicester",
              },
              Object {
                "id": 6,
                "team1": "C Palace",
                "team2": "Liverpool",
              },
              Object {
                "id": 7,
                "team1": "Brighton",
                "team2": "Leicester",
              },
            ]
          }
          onPlayed={[Function]}
          onPredict={[Function]}
          onToggleNext={[Function]}
          slideWidth={620}
          teams={
            Array [
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
                "draw": 9,
                "lose": 4,
                "played": 31,
                "points": 63,
                "team": "Liverpool",
                "win": 18,
              },
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
                "draw": 9,
                "lose": 15,
                "played": 31,
                "points": 30,
                "team": "C Palace",
                "win": 7,
              },
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
                "draw": 10,
                "lose": 12,
                "played": 30,
                "points": 34,
                "team": "Brighton",
                "win": 8,
              },
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
                "draw": 10,
                "lose": 10,
                "played": 30,
                "points": 40,
                "team": "Leicester",
                "win": 10,
              },
            ]
          }
        />
        <Slide
          group={
            Array [
              Object {
                "id": 8,
                "team1": "C Palace",
                "team2": "Liverpool",
              },
              Object {
                "id": 9,
                "team1": "Brighton",
                "team2": "Leicester",
              },
            ]
          }
          onPlayed={[Function]}
          onPredict={[Function]}
          onToggleNext={[Function]}
          slideWidth={620}
          teams={
            Array [
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
                "draw": 9,
                "lose": 4,
                "played": 31,
                "points": 63,
                "team": "Liverpool",
                "win": 18,
              },
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
                "draw": 9,
                "lose": 15,
                "played": 31,
                "points": 30,
                "team": "C Palace",
                "win": 7,
              },
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
                "draw": 10,
                "lose": 12,
                "played": 30,
                "points": 34,
                "team": "Brighton",
                "win": 8,
              },
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
                "draw": 10,
                "lose": 10,
                "played": 30,
                "points": 40,
                "team": "Leicester",
                "win": 10,
              },
            ]
          }
        />
      </glamorous(div)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Slide
            group={
              Array [
                Object {
                  "id": 0,
                  "team1": "C Palace",
                  "team2": "Liverpool",
                },
                Object {
                  "id": 1,
                  "team1": "Brighton",
                  "team2": "Leicester",
                },
                Object {
                  "id": 2,
                  "team1": "C Palace",
                  "team2": "Liverpool",
                },
                Object {
                  "id": 3,
                  "team1": "Brighton",
                  "team2": "Leicester",
                },
                Object {
                  "id": 4,
                  "team1": "C Palace",
                  "team2": "Liverpool",
                },
                Object {
                  "id": 5,
                  "team1": "Brighton",
                  "team2": "Leicester",
                },
                Object {
                  "id": 6,
                  "team1": "C Palace",
                  "team2": "Liverpool",
                },
                Object {
                  "id": 7,
                  "team1": "Brighton",
                  "team2": "Leicester",
                },
              ]
            }
            onPlayed={[Function]}
            onPredict={[Function]}
            onToggleNext={[Function]}
            slideWidth={620}
            teams={
              Array [
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
                  "draw": 9,
                  "lose": 4,
                  "played": 31,
                  "points": 63,
                  "team": "Liverpool",
                  "win": 18,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
                  "draw": 9,
                  "lose": 15,
                  "played": 31,
                  "points": 30,
                  "team": "C Palace",
                  "win": 7,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
                  "draw": 10,
                  "lose": 12,
                  "played": 30,
                  "points": 34,
                  "team": "Brighton",
                  "win": 8,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
                  "draw": 10,
                  "lose": 10,
                  "played": 30,
                  "points": 40,
                  "team": "Leicester",
                  "win": 10,
                },
              ]
            }
          />,
          <Slide
            group={
              Array [
                Object {
                  "id": 8,
                  "team1": "C Palace",
                  "team2": "Liverpool",
                },
                Object {
                  "id": 9,
                  "team1": "Brighton",
                  "team2": "Leicester",
                },
              ]
            }
            onPlayed={[Function]}
            onPredict={[Function]}
            onToggleNext={[Function]}
            slideWidth={620}
            teams={
              Array [
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
                  "draw": 9,
                  "lose": 4,
                  "played": 31,
                  "points": 63,
                  "team": "Liverpool",
                  "win": 18,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
                  "draw": 9,
                  "lose": 15,
                  "played": 31,
                  "points": 30,
                  "team": "C Palace",
                  "win": 7,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
                  "draw": 10,
                  "lose": 12,
                  "played": 30,
                  "points": 34,
                  "team": "Brighton",
                  "win": 8,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
                  "draw": 10,
                  "lose": 10,
                  "played": 30,
                  "points": 40,
                  "team": "Leicester",
                  "win": 10,
                },
              ]
            }
          />,
        ],
        "currentPos": 0,
        "width": 1240,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "group": Array [
              Object {
                "id": 0,
                "team1": "C Palace",
                "team2": "Liverpool",
              },
              Object {
                "id": 1,
                "team1": "Brighton",
                "team2": "Leicester",
              },
              Object {
                "id": 2,
                "team1": "C Palace",
                "team2": "Liverpool",
              },
              Object {
                "id": 3,
                "team1": "Brighton",
                "team2": "Leicester",
              },
              Object {
                "id": 4,
                "team1": "C Palace",
                "team2": "Liverpool",
              },
              Object {
                "id": 5,
                "team1": "Brighton",
                "team2": "Leicester",
              },
              Object {
                "id": 6,
                "team1": "C Palace",
                "team2": "Liverpool",
              },
              Object {
                "id": 7,
                "team1": "Brighton",
                "team2": "Leicester",
              },
            ],
            "onPlayed": [Function],
            "onPredict": [Function],
            "onToggleNext": [Function],
            "slideWidth": 620,
            "teams": Array [
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
                "draw": 9,
                "lose": 4,
                "played": 31,
                "points": 63,
                "team": "Liverpool",
                "win": 18,
              },
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
                "draw": 9,
                "lose": 15,
                "played": 31,
                "points": 30,
                "team": "C Palace",
                "win": 7,
              },
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
                "draw": 10,
                "lose": 12,
                "played": 30,
                "points": 34,
                "team": "Brighton",
                "win": 8,
              },
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
                "draw": 10,
                "lose": 10,
                "played": 30,
                "points": 40,
                "team": "Leicester",
                "win": 10,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "group": Array [
              Object {
                "id": 8,
                "team1": "C Palace",
                "team2": "Liverpool",
              },
              Object {
                "id": 9,
                "team1": "Brighton",
                "team2": "Leicester",
              },
            ],
            "onPlayed": [Function],
            "onPredict": [Function],
            "onToggleNext": [Function],
            "slideWidth": 620,
            "teams": Array [
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
                "draw": 9,
                "lose": 4,
                "played": 31,
                "points": 63,
                "team": "Liverpool",
                "win": 18,
              },
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
                "draw": 9,
                "lose": 15,
                "played": 31,
                "points": 30,
                "team": "C Palace",
                "win": 7,
              },
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
                "draw": 10,
                "lose": 12,
                "played": 30,
                "points": 34,
                "team": "Brighton",
                "win": 8,
              },
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
                "draw": 10,
                "lose": 10,
                "played": 30,
                "points": 40,
                "team": "Leicester",
                "win": 10,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <glamorous(div)
          currentPos={0}
          width={1240}
        >
          <Slide
            group={
              Array [
                Object {
                  "id": 0,
                  "team1": "C Palace",
                  "team2": "Liverpool",
                },
                Object {
                  "id": 1,
                  "team1": "Brighton",
                  "team2": "Leicester",
                },
                Object {
                  "id": 2,
                  "team1": "C Palace",
                  "team2": "Liverpool",
                },
                Object {
                  "id": 3,
                  "team1": "Brighton",
                  "team2": "Leicester",
                },
                Object {
                  "id": 4,
                  "team1": "C Palace",
                  "team2": "Liverpool",
                },
                Object {
                  "id": 5,
                  "team1": "Brighton",
                  "team2": "Leicester",
                },
                Object {
                  "id": 6,
                  "team1": "C Palace",
                  "team2": "Liverpool",
                },
                Object {
                  "id": 7,
                  "team1": "Brighton",
                  "team2": "Leicester",
                },
              ]
            }
            onPlayed={[Function]}
            onPredict={[Function]}
            onToggleNext={[Function]}
            slideWidth={620}
            teams={
              Array [
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
                  "draw": 9,
                  "lose": 4,
                  "played": 31,
                  "points": 63,
                  "team": "Liverpool",
                  "win": 18,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
                  "draw": 9,
                  "lose": 15,
                  "played": 31,
                  "points": 30,
                  "team": "C Palace",
                  "win": 7,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
                  "draw": 10,
                  "lose": 12,
                  "played": 30,
                  "points": 34,
                  "team": "Brighton",
                  "win": 8,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
                  "draw": 10,
                  "lose": 10,
                  "played": 30,
                  "points": 40,
                  "team": "Leicester",
                  "win": 10,
                },
              ]
            }
          />
          <Slide
            group={
              Array [
                Object {
                  "id": 8,
                  "team1": "C Palace",
                  "team2": "Liverpool",
                },
                Object {
                  "id": 9,
                  "team1": "Brighton",
                  "team2": "Leicester",
                },
              ]
            }
            onPlayed={[Function]}
            onPredict={[Function]}
            onToggleNext={[Function]}
            slideWidth={620}
            teams={
              Array [
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
                  "draw": 9,
                  "lose": 4,
                  "played": 31,
                  "points": 63,
                  "team": "Liverpool",
                  "win": 18,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
                  "draw": 9,
                  "lose": 15,
                  "played": 31,
                  "points": 30,
                  "team": "C Palace",
                  "win": 7,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
                  "draw": 10,
                  "lose": 12,
                  "played": 30,
                  "points": 34,
                  "team": "Brighton",
                  "win": 8,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
                  "draw": 10,
                  "lose": 10,
                  "played": 30,
                  "points": 40,
                  "team": "Leicester",
                  "win": 10,
                },
              ]
            }
          />
        </glamorous(div)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Slide
              group={
                Array [
                  Object {
                    "id": 0,
                    "team1": "C Palace",
                    "team2": "Liverpool",
                  },
                  Object {
                    "id": 1,
                    "team1": "Brighton",
                    "team2": "Leicester",
                  },
                  Object {
                    "id": 2,
                    "team1": "C Palace",
                    "team2": "Liverpool",
                  },
                  Object {
                    "id": 3,
                    "team1": "Brighton",
                    "team2": "Leicester",
                  },
                  Object {
                    "id": 4,
                    "team1": "C Palace",
                    "team2": "Liverpool",
                  },
                  Object {
                    "id": 5,
                    "team1": "Brighton",
                    "team2": "Leicester",
                  },
                  Object {
                    "id": 6,
                    "team1": "C Palace",
                    "team2": "Liverpool",
                  },
                  Object {
                    "id": 7,
                    "team1": "Brighton",
                    "team2": "Leicester",
                  },
                ]
              }
              onPlayed={[Function]}
              onPredict={[Function]}
              onToggleNext={[Function]}
              slideWidth={620}
              teams={
                Array [
                  Object {
                    "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
                    "draw": 9,
                    "lose": 4,
                    "played": 31,
                    "points": 63,
                    "team": "Liverpool",
                    "win": 18,
                  },
                  Object {
                    "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
                    "draw": 9,
                    "lose": 15,
                    "played": 31,
                    "points": 30,
                    "team": "C Palace",
                    "win": 7,
                  },
                  Object {
                    "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
                    "draw": 10,
                    "lose": 12,
                    "played": 30,
                    "points": 34,
                    "team": "Brighton",
                    "win": 8,
                  },
                  Object {
                    "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
                    "draw": 10,
                    "lose": 10,
                    "played": 30,
                    "points": 40,
                    "team": "Leicester",
                    "win": 10,
                  },
                ]
              }
            />,
            <Slide
              group={
                Array [
                  Object {
                    "id": 8,
                    "team1": "C Palace",
                    "team2": "Liverpool",
                  },
                  Object {
                    "id": 9,
                    "team1": "Brighton",
                    "team2": "Leicester",
                  },
                ]
              }
              onPlayed={[Function]}
              onPredict={[Function]}
              onToggleNext={[Function]}
              slideWidth={620}
              teams={
                Array [
                  Object {
                    "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
                    "draw": 9,
                    "lose": 4,
                    "played": 31,
                    "points": 63,
                    "team": "Liverpool",
                    "win": 18,
                  },
                  Object {
                    "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
                    "draw": 9,
                    "lose": 15,
                    "played": 31,
                    "points": 30,
                    "team": "C Palace",
                    "win": 7,
                  },
                  Object {
                    "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
                    "draw": 10,
                    "lose": 12,
                    "played": 30,
                    "points": 34,
                    "team": "Brighton",
                    "win": 8,
                  },
                  Object {
                    "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
                    "draw": 10,
                    "lose": 10,
                    "played": 30,
                    "points": 40,
                    "team": "Leicester",
                    "win": 10,
                  },
                ]
              }
            />,
          ],
          "currentPos": 0,
          "width": 1240,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "group": Array [
                Object {
                  "id": 0,
                  "team1": "C Palace",
                  "team2": "Liverpool",
                },
                Object {
                  "id": 1,
                  "team1": "Brighton",
                  "team2": "Leicester",
                },
                Object {
                  "id": 2,
                  "team1": "C Palace",
                  "team2": "Liverpool",
                },
                Object {
                  "id": 3,
                  "team1": "Brighton",
                  "team2": "Leicester",
                },
                Object {
                  "id": 4,
                  "team1": "C Palace",
                  "team2": "Liverpool",
                },
                Object {
                  "id": 5,
                  "team1": "Brighton",
                  "team2": "Leicester",
                },
                Object {
                  "id": 6,
                  "team1": "C Palace",
                  "team2": "Liverpool",
                },
                Object {
                  "id": 7,
                  "team1": "Brighton",
                  "team2": "Leicester",
                },
              ],
              "onPlayed": [Function],
              "onPredict": [Function],
              "onToggleNext": [Function],
              "slideWidth": 620,
              "teams": Array [
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
                  "draw": 9,
                  "lose": 4,
                  "played": 31,
                  "points": 63,
                  "team": "Liverpool",
                  "win": 18,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
                  "draw": 9,
                  "lose": 15,
                  "played": 31,
                  "points": 30,
                  "team": "C Palace",
                  "win": 7,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
                  "draw": 10,
                  "lose": 12,
                  "played": 30,
                  "points": 34,
                  "team": "Brighton",
                  "win": 8,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
                  "draw": 10,
                  "lose": 10,
                  "played": 30,
                  "points": 40,
                  "team": "Leicester",
                  "win": 10,
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "group": Array [
                Object {
                  "id": 8,
                  "team1": "C Palace",
                  "team2": "Liverpool",
                },
                Object {
                  "id": 9,
                  "team1": "Brighton",
                  "team2": "Leicester",
                },
              ],
              "onPlayed": [Function],
              "onPredict": [Function],
              "onToggleNext": [Function],
              "slideWidth": 620,
              "teams": Array [
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
                  "draw": 9,
                  "lose": 4,
                  "played": 31,
                  "points": 63,
                  "team": "Liverpool",
                  "win": 18,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
                  "draw": 9,
                  "lose": 15,
                  "played": 31,
                  "points": 30,
                  "team": "C Palace",
                  "win": 7,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
                  "draw": 10,
                  "lose": 12,
                  "played": 30,
                  "points": 34,
                  "team": "Brighton",
                  "win": 8,
                },
                Object {
                  "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
                  "draw": 10,
                  "lose": 10,
                  "played": 30,
                  "points": 40,
                  "team": "Leicester",
                  "win": 10,
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
